HubNotificationHandler/src/main/java/com/eagle/hubnotifier/service/impl/NotifyHookServiceImpl.java

```diff
@@ -162,15 +166,27 @@ private void handleTopicReplyEvent(Map<String, Object> data) {
         nEvent.setEventId(Constants.DISCUSSION_REPLY_EVENT_ID);
         Map<String, Object> tagValues = new HashMap<>();
         tagValues.put(Constants.COMMENT_TAG, ((List<String>) data.get(Constants.PARAM_CONTENT_CONSTANT)).get(0));
-        HubUser repliedByUser = userRepository.findByKey(Constants.USER_ROLE + ":" + ((List<String>)data.get(Constants.PARAM_UID)).get(0));
+        HubUser repliedByUser = userRepository.findByKey(Constants.USER_ROLE + ":" + ((List<String>) data.get(Constants.PARAM_UID)).get(0));
         tagValues.put(Constants.COMMENTED_BY_NAME_TAG, repliedByUser.getUsername());
         tagValues.put(Constants.DISCUSSION_CREATION_TARGET_URL, configuration.getDiscussionCreateUrl() + ((List<String>) data.get(Constants.PARAM_TID)).get(0));
         Map<String, List<String>> recipients = new HashMap<>();
-        HubTopic topicData = hubTopicRepository.findByKey(Constants.TOPIC_VALUE_CONSTANTS + ":" + ((List<String>)data.get(Constants.PARAM_TID)).get(0));
+        HubTopic topicData = hubTopicRepository.findByKey(Constants.TOPIC_VALUE_CONSTANTS + ":" + ((List<String>) data.get(Constants.PARAM_TID)).get(0));
         if (!ObjectUtils.isEmpty(topicData)) {
             tagValues.put(Constants.DISCUSSION_CREATION_TOPIC_TAG, topicData.getTitle());
             HubUser author = userRepository.findByKey(Constants.USER_ROLE + ":" + topicData.getUid());
-            recipients.put(Constants.AUTHOR_ROLE, Arrays.asList(author.getUsername()));
+            List<String> watchersList = getWatchersList(String.valueOf(topicData.getTid()));
```

Comment:
This block of code is added in all the methods... 
Can we move this to private method and simply pass the required object reference to update the list ?